packages/realm/src/app-services/SyncSession.ts

```diff
@@ -53,13 +53,29 @@ export enum ProgressMode {
   ForCurrentlyOutstandingWork = "forCurrentlyOutstandingWork",
 }
 
-export type ProgressNotificationCallback =
+/** @deprecated */
+export type PartitionBasedSyncProgressNotificationCallback =
```

Comment:
For the new public types that got introduced, they also need to be exported to users similar to [this](https://github.com/realm/realm-js/blob/a72bbc2c13ca2afc99b438ffc0aa207351c6b101/packages/realm/src/Realm.ts#L1326).

For deprecated ones such as the PBS-specific one, we list them below [this line](https://github.com/realm/realm-js/blob/a72bbc2c13ca2afc99b438ffc0aa207351c6b101/packages/realm/src/Realm.ts#L1369).